reporters/kamon-prometheus/src/main/scala/kamon/prometheus/ScrapeDataBuilder.scala

```diff
@@ -126,6 +166,25 @@ class ScrapeDataBuilder(prometheusConfig: PrometheusSettings.Generic, environmen
       }
     )
 
+  private def appendSummaryQuantiles(name: String,
+                                     tags: TagSet,
+                                     distribution: Distribution,
+                                     unit: MeasurementUnit,
+                                     quantiles: Seq[java.lang.Double]): Unit = {
+    val percentileIter = distribution.percentilesIterator
+    val percentiles = quantiles.sorted.map { quant =>
```

Comment:
It would be simpler to call `Distribution.percentile(...)` to get those values out. Under the hood it kind of does the same you are doing here.

Reply:
That was my first implementation. To my understanding, calling Distribution.percentile(...) would create an iterator for every percentile that is requested and iterate through it until the first is reached that is >= the wanted. 
This would result in iterating over the same percentiles multiple times. Thats why i thought creating one iterator and using it for all percentiles might be better for performance

Reply:
Indeed, your code will perform better!